File: codes/DoubleHashTable/Java/DoubleHashTable/Slot.java
package DoubleHashTable;

public class Slot<K, T> {
    final static int MAX_TBL = 100;
    enum SlotStatus { EMPTY, DELETED, INUSE };

    private K key;
    private T value;
    private SlotStatus status;

    public void setKey(K k) {
        this.key = k;
    }

    public K getKey() {
        return this.key;
    }

    public void setvalue(T v) {
        this.value = v;
    }

    public T getValue() {
        return this.value;
    }

    public void setStatus(SlotStatus ss) {
        this.status = ss;
    }

    public SlotStatus getStatus() {
        return this.status;
    }
}

File: codes/DoubleHashTable/Python/Slot.py
from enum import Enum

class SlotStatus(Enum):
    EMPTY = 0
    DELETED = 1
    INUSE = 2

class Slot:
    MAX_TBL = 100

    def __init__(self):
        self.key = None
        self.value = None
        self.status = None
